core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -432,6 +436,18 @@ private DefaultFunctionResolver minute() {
     );
   }
 
+  /**
+   * MINUTE(STRING/TIME/DATETIME/TIMESTAMP). return the minute value for time.
+   */
+  private DefaultFunctionResolver minute_of_day() {
+    return define(BuiltinFunctionName.MINUTE_OF_DAY.getName(),
+        impl(nullMissingHandling(DateTimeFunction::exprMinuteOfDay), INTEGER, STRING),
+        impl(nullMissingHandling(DateTimeFunction::exprMinuteOfDay), INTEGER, TIME),
```

Comment:
You can accept `DATE` and return 0 for any value of such type.

Reply:
Proposed fix (see below will do this automatically)

Reply:
Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/189/commits/956cde1cbda5d8f787fa926f1d2279359470d82a